components/user-mgt/org.wso2.carbon.identity.user.pre.update.password.action/src/main/java/org/wso2/carbon/identity/user/pre/update/password/action/model/PasswordUpdatingUser.java

```diff
@@ -0,0 +1,132 @@
+
+import java.security.Key;
+import java.security.cert.X509Certificate;
+import java.security.interfaces.RSAPublicKey;
+
+/**
+ * This class models the user object at a pre update password trigger.
+ * UserRequest is the entity that represents the user object that is sent to Action
+ * over {@link org.wso2.carbon.identity.action.execution.model.ActionExecutionRequest}.
+ */
+public class PasswordUpdatingUser extends User {
+
+    private final Object updatingCredential;
+
+    private PasswordUpdatingUser(Builder builder) {
+
+
+        super(builder.id);
+        this.updatingCredential = builder.updatingCredential;
+    }
+
+    public Object getUpdatingCredential() {
+
+        return updatingCredential;
+    }
+
+    /**
+     * Builder for TokenRequest.
+     */
+    public static class Builder {
+
+        private String id;
+        private Object updatingCredential;
+        private Credential unEncryptedCredential;
+        private boolean isCredentialEncryptionRequired = false;
+        private X509Certificate certificate;
+
+        public Builder() {
+
+        }
+
+        public Builder id(String id) {
+
+            this.id = id;
+            return this;
+        }
+
+        public Builder updatingCredential(Credential unEncryptedCredential,
+                                          boolean isCredentialEncryptionRequired, X509Certificate certificate) {
+
+            this.unEncryptedCredential = unEncryptedCredential;
+            this.isCredentialEncryptionRequired = isCredentialEncryptionRequired;
+            this.certificate = certificate;
+            return this;
+        }
+
+        public PasswordUpdatingUser build() throws ActionExecutionRequestBuilderException {
+
+            if (!isCredentialEncryptionRequired) {
+                this.updatingCredential = unEncryptedCredential;
+                return new PasswordUpdatingUser(this);
+            }
+
+            if (certificate == null) {
+                throw new IllegalStateException("Certificate is required for encryption.");
+            }
+
+            this.updatingCredential = encryptCredential(unEncryptedCredential, certificate);
+            return new PasswordUpdatingUser(this);
+        }
+
+        private String encryptCredential(Credential unEncryptedCredential, X509Certificate certificate)
+                throws ActionExecutionRequestBuilderException {
+
+            try {
+                // Extract the public key from the certificate
+                Key publicKey = certificate.getPublicKey();
+
+                // Serialize the object as a JSON string using Gson
+                String credentialJson = new Gson().toJson(unEncryptedCredential);
+
+                // Create a JWE header with encryption details
+                JWEHeader header = new JWEHeader.Builder(JWEAlgorithm.RSA_OAEP_256, EncryptionMethod.A256GCM)
+                        .contentType("application/json") // Specify content type as JSON
+                        .build();
```

Comment:
Do we have any config to get default encryption alogos and mechanisms ?
I recall such. Let's check on this.
We can get this merged and look at this as a new task

Reply:
Will check on this later